pillar/vector/mitx.sls

```diff
@@ -0,0 +1,371 @@
+        inputs:
+          - cms_stderr_log_parser
+        type: add_fields
+        fields:
+          labels:
+            - edx_cms_stderr
+
+      cms_stderr_timestamp_renamer:
+        inputs:
+          - cms_stderr_log_labeler
+        type: rename_fields
+        fields:
+          time: "@timestamp"
+
+      cms_stderr_sampler:
+        inputs:
+          - cms_stderr_timestamp_renamer
+        type: sampler
+        rate: 1
+        exclude:
+          type: check_fields
+          "message.starts_with": "GET"
+
+      # the following transform also has to process lines with varying formats.
+      # examples:
+      # 1)
+      #     [2021-02-25 19:09:22 +0000] [2056637] [INFO] POST /search/course_discovery/
+      # 2)
+      #     2021-02-25 19:09:22,728 WARNING 2056637 [django.security.csrf] [user None] [ip 127.0.0.1] log.py:222 - Forbidden (CSRF cookie not set.): /search/course_discovery/
+      lms_stderr_log_parser:
+        inputs:
+          - lms_stderr_log
+        type: regex_parser
+        field: message
+        overwrite_target: true
+        patterns:
+          - '\[(?P<time>\d{4}-\d{2}-\d{2} \d{2}:\d{2}:\d{2}) [+\-]\d{4}\] \[(?P<pid>\d+)\] \[(?P<log_level>[A-Z]+)\] (?P<message>.*)'
+          - '(?P<time>\d{4}-\d{2}-\d{2} \d{2}:\d{2}:\d{2}),\d{3} (?P<log_level>[A-Z]+) (?P<pid>\d+) (?P<message>.*)'
+        types:
+          time: timestamp|%Y-%m-%d %H:%M:%S
+          pid: bytes
+
+      lms_stderr_log_labeler:
+        inputs:
+          - lms_stderr_log_parser
+        type: add_fields
+        fields:
+          labels:
+            - edx_lms_stderr
+
+      lms_stderr_timestamp_renamer:
+        inputs:
+          - lms_stderr_log_labeler
+        type: rename_fields
+        fields:
+          time: "@timestamp"
+
+      lms_stderr_sampler:
+        inputs:
+          - lms_stderr_timestamp_renamer
+        type: sampler
+        rate: 1
+        exclude:
+          type: check_fields
+          "message.starts_with": "GET"
+
+      {% endif %}
+
+      {% if 'edx-worker' in salt.grains.get('roles') %}
+
+      # edX worker's "cms" supervisor stderr log,
+      # e.g. /edx/var/log/supervisor/cms_default_4-stderr.log
+      # is formatted as in these examples:
+      #
+      # [2021-02-25 16:29:59,191: ERROR/ForkPoolWorker-4] git_auto_export.tasks.async_export_to_git[b6320d3b-d208-4ea4-ae53-4d0fbedc247c]: Failed async course content export to git (course id: course-v1:MITx+8.011r_8+2021_Spring): Unable to push changes.  This is usually because the remote repository cannot be contacted
+      # [2021-02-25 18:05:02,524: INFO/MainProcess] Received task: lms.djangoapps.discussion.tasks.update_discussions_map[9413651f-6863-4f62-bc61-85f565515568]  ETA:[2021-02-25 18:05:32.505810+00:00]
+      # [2021-02-25 16:31:14,432: ERROR/ForkPoolWorker-4] Error running git push command: b"To github.mit.edu:MITx-Studio2LMS/content-mit-8011r_8-2021_Spring.git\n ! [rejected]  [...]
+      #
+      worker_cms_stderr_log_parser:
```

Comment:
nitpick: Should we call this edx_worker.. given that the role is `edx-worker`

Reply:
I'd removed "edx_" from the beginning of all of the `source`s because in the context of this file we know that they're edx-related, and the IDs are unique within the file.